### PR TITLE
[opensubdiv] New port

### PR DESCRIPTION
**Pull request by ousttrue**
**Describe the pull request**

> What does your PR fix? #5310

Adds a port for the the https://github.com/PixarAnimationStudios/OpenSubdiv

> Which triplets are supported/not supported? Have you updated the CI baseline?

Tested x64-Windows.

> Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@ousttrue, thanks for the PR!

The CI failed with following failure, could you format the vcpkg.json file with command 

 `./vcpkg format-manifest ports/opensubdiv/vcpkg.json`

Failures:
The formatting of the manifest files didn't match our expectation.
See github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#manifest for solution.

File: ports/opensubdiv/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "opensubdiv",
+  "version-string": "3.4.3",
+  "homepage": "https://github.com/PixarAnimationStudios/OpenSubdiv",
+  "description": "An Open-Source subdivision surface library.",
+  "dependencies": []
```
Comment (PhoebeHui) [inline ports/opensubdiv/vcpkg.json:6]:
```suggestion
  "description": "An Open-Source subdivision surface library."
```

File: ports/opensubdiv/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/bin")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+
+# # Handle copyright
+file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/opensubdiv RENAME copyright)
+
+# # Post-build test for cmake libraries
+# vcpkg_test_cmake(PACKAGE_NAME opensubdiv)
```
Comment (PhoebeHui) [inline ports/opensubdiv/portfile.cmake:44]:
```suggestion
file(INSTALL ${SOURCE_PATH}/LICENSE.txt DESTINATION ${CURRENT_PACKAGES_DIR}/share/${PORT} RENAME copyright)

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

File: ports/opensubdiv/vcpkg.json (hunk 1 above)
Comment (ousttrue) [reply ports/opensubdiv/vcpkg.json]: I understand ./vcpkg format-manifest

File: ports/opensubdiv/portfile.cmake (hunk 2 above)
Comment (ousttrue) [reply ports/opensubdiv/portfile.cmake]: Thank you. I fixed it.

**Comment by PhoebeHui** [comment]
@ousttrue, here is the failure for each triplets in CI testing.
[install-arm64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/5274896/install-arm64-windows-dbg-out.log)
[install-x64-windows-static-rel-out.log](https://github.com/microsoft/vcpkg/files/5274898/install-x64-windows-static-rel-out.log)
[install-x86-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/5274906/install-x86-windows-dbg-out.log)
[install-x64-uwp-dbg-out.log](https://github.com/microsoft/vcpkg/files/5274914/install-x64-uwp-dbg-out.log)

```
x64-windows-static:
LINK : warning LNK4044: unrecognized option '/openmp'; ignored
libcpmt.lib(wlocale.obj) : error LNK2001: unresolved external symbol _purecall
libcpmt.lib(xlocale.obj) : error LNK2001: unresolved external symbol _purecall
main.cpp.obj : error LNK2001: unresolved external symbol _purecall
libcpmt.lib(locale0.obj) : error LNK2001: unresolved external symbol _purecall
libcpmt.lib(cerr.obj) : error LNK2001: unresolved external symbol _purecall
libcpmt.lib(locale.obj) : error LNK2001: unresolved external symbol _purecall
libcpmt.lib(wlocale.obj) : error LNK2001: unresolved external symbol "void * __cdecl operator new(unsigned __int64)" (??2@YAPEAX_K@Z)
libcpmt.lib(xlocale.obj) : error LNK2001: unresolved external symbol "void * __cdecl operator new(unsigned __int64)" (??2@YAPEAX_K@Z)
main.cpp.obj : error LNK2001: unresolved external symbol "void * __cdecl operator new(unsigned __int64)" (??2@YAPEAX_K@Z)

x86-windows: 
nvcc fatal   : 32 bit compilation is only supported for Microsoft Visual Studio 2013 and earlier
CMake Error at osd_static_gpu_generated_cudaKernel.cu.obj.Debug.cmake:220 (message):

arm64-windows:
This version of D:\buildtrees\opensubdiv\arm64-windows-dbg\bin\stringify.exe is not compatible with the version of Windows you're running. Check your computer's system information and then contact the software publisher.

arm-uwp and x64-uwp:
error C3861: 'LoadLibraryW': identifier not found
```

**Comment by ousttrue** [comment]
👍

File: ports/opensubdiv/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "name": "opensubdiv",
+  "version-string": "3.4.3",
+  "description": "An Open-Source subdivision surface library.",
+  "homepage": "https://github.com/PixarAnimationStudios/OpenSubdiv",
+  "supports": "windows & !arm & !uwp & !static"
```
Comment (ras0219-msft) [inline ports/opensubdiv/vcpkg.json:6]:
```suggestion
  "supports": "windows & !arm & !uwp"
```
`static` means static library linkage, not CRT linkage. There is not currently a static CRT linkage tag, so you'll need to add those failures to the ci.baseline.txt file.

File: ports/opensubdiv/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        -DNO_TESTS=ON
+)
+
+vcpkg_install_cmake()
+
+# # Moves all .cmake files from /debug/share/opensubdiv/ to /share/opensubdiv/
+# # See /docs/maintainers/vcpkg_fixup_cmake_targets.md for more details
+# vcpkg_fixup_cmake_targets(CONFIG_PATH cmake TARGET_PATH share/opensubdiv)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/bin")
```
Comment (ras0219-msft) [inline ports/opensubdiv/portfile.cmake:37]: Why is this needed? If the library only builds statically, then there shouldn't be any files in the bin folder.

Otherwise, if it is able to produce DLLs, it should probably not be `ONLY_STATIC_LIBRARY`.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: ports/opensubdiv/portfile.cmake (hunk 4 above)
Comment (ousttrue) [reply ports/opensubdiv/portfile.cmake:37]: This is remove bin/stringify.exe.
Stringify may be used during the build process.
There was no option not to make.

**Comment by ousttrue** [comment]
I'm not sure about targets other than x64-windows...

**Comment by ousttrue** [comment]
static build fixed. 

CI x86_windows #13722

**Review by PhoebeHui — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!
